### DISCUSSION TITLE
Multiple libraries using codegen

### DISCUSSION THREAD

**Discussion by riteshshukla04** [reactwg/react-native-new-architecture#241] [Q&A] [answered]
The latest codegen configuration doesnt have option to add multiple libraries (Like before) . How can we add multiple libraries or now is it limited to just one.

**Comment by cortinico** [comment]
> The latest codegen configuration doesnt have option to add multiple libraries (Like before)

You'll have to create separate NPM packages if you intend to do so. Also, you can still add multiple components/modules to the same library if you wish.

**Reply by riteshshukla04** [reply]
My use case was just  to create few in app native module and creating npm package would be bit of overkill. Just curious, any particular reason that it is limited to one?

**Reply by cortinico** [reply, marked as answer]
> Just curious, any particular reason that it is limited to one?

Because on Android we need to link against a static/object library which contains all the C++ code. If we were to allow more than one library, we'll have to add more complexity in the build logic so that there will be multiple CMake targets. This will also complicate the autolinking which is already quite involved, so we opted for having 1 library per 1 NPM module
